app/models/household_good.rb

```diff
@@ -0,0 +1,12 @@
+class HouseholdGood < ApplicationRecord
+  validates :weight, :category, presence: true
+  validates :name, presence: true, uniqueness: { scope: :category }
+
+  def key
+    @key ||= name.gsub(/[^0-9a-z]/i, ' ').strip.gsub(/\s+/, '_').downcase
+  end
+
+  def weight_key
+    @weight_key ||= key + '_weight'
```

Comment:
There's a thousand different ways to do string interpolation in Ruby, so don't feel compelled to change this. But… you _could_ do:

```rb
@weight_key ||= "#{key}_weight"
```